thirdparty/CMakeLists.txt

```diff
@@ -1 +1,36 @@
-add_subdirectory(spdlog)
+include(FetchContent)
+
+###################################################################################################
+# - cnmem -----------------------------------------------------------------------------------------
+
+FetchContent_Declare(
+    cnmem
+    GIT_REPOSITORY https://github.com/NVIDIA/cnmem
+    GIT_TAG        37896cc9bfc6536a8c878a1e675835c22d827821
+    GIT_SHALLOW    true
+)
+
+FetchContent_GetProperties(cnmem)
+if(NOT cnmem_POPULATED)
+  FetchContent_Populate(cnmem)
+  set(CNMEM_INCLUDE_DIR "${cnmem_SOURCE_DIR}/include" PARENT_SCOPE)
+  set(CNMEM_SOURCE_DIR "${cnmem_SOURCE_DIR}/src" PARENT_SCOPE)
+  # We are not using the cnmem CMake targets, so no need to call `add_subdirectory()`.
+endif()
+
+###################################################################################################
+# - spdlog ----------------------------------------------------------------------------------------
+
+FetchContent_Declare(
+    spdlog
+    GIT_REPOSITORY https://github.com/gabime/spdlog.git
+    GIT_TAG        v1.6.0
+    GIT_SHALLOW    true
+)
+
+FetchContent_GetProperties(spdlog)
+if(NOT spdlog_POPULATED)
+  FetchContent_Populate(spdlog)
+  set(SPDLOG_INCLUDE_DIR "${spdlog_SOURCE_DIR}/include" PARENT_SCOPE)
```

Comment:
Same here.

Reply:
Here calling `FetchContent_MakeAvailable` causes spdlog static lib to be built, which we don't really need.